imucapture/src/main.cpp

```diff
@@ -76,245 +164,178 @@ int kbhit (){
+		<< " opened." << std::endl;
+
+	// Note the differences between MTix and MTmk4
+	std::cout << "Configuring the device..." << std::endl;
+	if (mtPort.deviceId().isMt9c() || mtPort.deviceId().isLegacyMtig())
+	{
+		// output orientation data
+		XsOutputMode outputMode = XOM_Orientation;
+
+		// output orientation data as quaternion
+		XsOutputSettings outputSettings = XOS_OrientationMode_Quaternion;
+
+		// set the device configuration
+		if (!device.setDeviceMode(outputMode, outputSettings))
+			throw std::runtime_error("Could not configure MT device. \
+				Aborting.");
+			
+	}
+	else if (mtPort.deviceId().isMtMk4() || mtPort.deviceId().isFmt_X000())
+	{
+		XsOutputConfiguration quat(XDI_Quaternion, 100);
+		XsOutputConfigurationArray configArray;
+		configArray.push_back(
+			XsOutputConfiguration(XDI_LatLon | XDI_SubFormatDouble, 100)
+		);
+		configArray.push_back(
+			XsOutputConfiguration(XDI_AltitudeEllipsoid, 100)
+		);
+		configArray.push_back(
+			XsOutputConfiguration(XDI_Quaternion | XDI_CoordSysNed, 100)
+		);
+		configArray.push_back(
+			XsOutputConfiguration(XDI_VelocityXYZ | XDI_CoordSysNed, 100)
+		);
+
+		configArray.push_back(quat);
+
+		if (!device.setOutputConfiguration(configArray))
+			throw std::runtime_error("Could not configure MTmk4 device. \
+				Aborting.");
+	}
+	else
+	{
+		throw std::runtime_error("Unknown device while configuring. Aborting.");
+	}
 
-			XsPortInfo portInfo(portName, XsBaud::numericToRate(baudRate));
-			portInfoArray.push_back(portInfo);
-		}
+	std::cout << "Putting device into measurement mode..." << std::endl;
+	if (!device.gotoMeasurement()) {
+		throw std::runtime_error("Could not put device into measurement mode. \
+			Aborting.");
+	}
+}
 
-		// Use the first detected device
-		auto mtPort = portInfoArray.at(0);
+XsPortInfo scan_usb_devices()
+{
+	std::cout << "Scanning for USB devices..." << std::endl;
+	XsPortInfoArray portInfoArray;
+	xsEnumerateUsbDevices(portInfoArray);
+	if (!portInfoArray.size()) // Can't find device
+	{
+		std::string portName;
+		int baudRate;
+		std::cout << "No USB Motion Tracker found." << std::endl << std::endl
+			<< "Enter COM port name (eg. /dev/ttyUSB0): " << std::endl;
+		std::cin >> portName;
+		std::cout << "Enter baud rate (eg. 115200): ";
+		std::cin >> baudRate;
+
+		XsPortInfo portInfo(portName, XsBaud::numericToRate(baudRate));
+		portInfoArray.push_back(portInfo);
+	}
 
-		std::cout << "Opening port..." << std::endl;
-		if (!device.openPort(mtPort)){
-			throw std::runtime_error("Could not open port. Aborting.");
-		}
-		
-		std::cout << "Putting device into configuration mode..." << std::endl;
-		if (!device.gotoConfig()) {
-			throw std::runtime_error("Could not put device into configuration mode. Aborting.");
-		}
+	// Use the first detected device
+	return portInfoArray.at(0);
+}
 
-		// Request the device Id to check the device type
-		mtPort.setDeviceId(device.getDeviceId());
+void set_message_to_packet(
```

Comment:
All parameters except the packet could be const.

Reply:
Got it.